metpy/calc/kinematics.py

```diff
@@ -432,3 +432,39 @@ def geostrophic_wind(heights, f, dx, dy):
     grad = _gradient(heights, *deltas)
     dy, dx = grad[-2:]  # Throw away unused gradient components
     return -norm_factor * dy, norm_factor * dx
+
+
+@exporter.export
+@check_units('[length]', '[temperature]')
+def montgomery_streamfunction(height, temperature):
+    r"""Compute the Montgomery Streamfunction on isentropic surfaces.
+
+    Parameters
+    ----------
+    height : array
```

Comment:
Instead of array, should probably be `pint.Quantity` (like [here](https://github.com/Unidata/MetPy/blob/master/metpy/calc/thermo.py#L32))